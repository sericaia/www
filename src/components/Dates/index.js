import React from 'react'
import format from 'date-fns/format'
import isArray from 'lodash/isArray'

const Dates = ({ date }) => {
  if (!isArray(date)) return null

  if (date.length === 2) {
    const from = format(new Date(date[0]), 'MMMM dd, yyyy')
    const to = format(new Date(date[1]), 'MMMM dd, yyyy')

    return (
      <>
        {from} to {to}
      </>
    )
  }

  if (date.length === 1) {
    const formattedDate = format(new Date(date), 'MMMM dd, yyyy')
    return formattedDate
  }

  return null // invalid
}

export default Dates
