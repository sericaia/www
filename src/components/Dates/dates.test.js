import React from 'react'
import { render } from '@testing-library/react'
import Dates from '.'

describe('Dates', () => {
  it('renders correctly one date', () => {
    const date = ['2019-09-12']
    const { container } = render(<Dates date={date} />)
    expect(container).toMatchSnapshot()
  })

  it('renders correctly from and to dates', () => {
    const date = ['2017-10-12', '2017-10-13']
    const { container } = render(<Dates date={date} />)
    expect(container).toMatchSnapshot()
  })
})
