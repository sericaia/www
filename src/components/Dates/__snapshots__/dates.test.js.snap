// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dates renders correctly from and to dates 1`] = `
<div>
  October 12, 2017
   to 
  October 13, 2017
</div>
`;

exports[`Dates renders correctly one date 1`] = `
<div>
  September 12, 2019
</div>
`;
